### ISSUE TITLE
Map not loading properly in a List View

### ISSUE THREAD

**Issue by RobinCPel** [fleaflet/flutter_map#623]
Hi! 

I am having some issues when trying to display multiple maps in a list view, demo: 
![flutter_map_test](https://user-images.githubusercontent.com/23524090/82670553-a695e180-9c3d-11ea-9094-1fa5d1832291.gif)

The app is supposed to show ten maps in a column, but the ones that are off screen are not loaded properly once they do move on screen. 

Source code of the demo: 

```dart
import 'package:flutter/material.dart';
import 'package:flutter_map/flutter_map.dart';
import 'package:latlong/latlong.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('🗺'), centerTitle: true),
      body: Center(
        child: ListView.builder(
          itemBuilder: (context, i) {
            if (i.isOdd) return Divider(height: 16.0);
            final index = i ~/ 2;
            if (index >= 10) return null;
            return _mapWidget();
          },
        ),
      ),
    );
  }

  Widget _mapWidget() {
    return Container(
      height: 150,
      child: FlutterMap(
        options: MapOptions(
          center: LatLng(51.993693, 6.718940),
          zoom: 12.0,
        ),
        layers: [
          TileLayerOptions(
            urlTemplate: "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
            subdomains: ['a', 'b', 'c'],
          ),
        ],
      ),
    );
  }
}

```

**Comment by bubinyang** [comment]
What are the versions of flutter and flutter_map?Do you need path_provider?When I run the project, I always report an error, version conflict

**Comment by RobinCPel** [comment]
## Flutter Version
<img width="759" alt="Screenshot 2020-05-27 at 16 10 54" src="https://user-images.githubusercontent.com/23524090/83031301-aed29000-a034-11ea-9336-a0d3e7599aac.png">

## Pubspec
<img width="605" alt="Screenshot 2020-05-27 at 16 12 56" src="https://user-images.githubusercontent.com/23524090/83031524-f1946800-a034-11ea-9262-6e9da4c293c1.png">

**Comment by bubinyang** [comment]
Thank you very much. My problem has been solved

**Comment by FilipeLarga** [comment]
I'm also having this issue in Android using an AnimatedListView with horizontal axis.

**Comment by FilipeLarga** [comment]
I fixed this by merging #637 on my own fork.

**Comment by RobinCPel** [comment]
Good find, I will try this soon!
